kernel/aster-nix/src/syscall/writev.rs

```diff
@@ -33,19 +20,21 @@ fn do_sys_writev(fd: FileDesc, io_vec_ptr: Vaddr, io_vec_count: usize) -> Result
         let filetable = current.file_table().lock();
         filetable.get_file(fd)?.clone()
     };
+
     let mut total_len = 0;
-    for i in 0..io_vec_count {
-        let io_vec = read_val_from_user::<IoVec>(io_vec_ptr + i * core::mem::size_of::<IoVec>())?;
-        if io_vec.base == 0 || io_vec.len == 0 {
+
+    let io_vecs = copy_iovs_from_user(io_vec_ptr, io_vec_count)?;
+    for io_vec in io_vecs.as_ref() {
+        if io_vec.is_empty() {
             continue;
         }
+
         let buffer = {
-            let base = io_vec.base;
-            let len = io_vec.len;
-            let mut buffer = vec![0u8; len];
-            read_bytes_from_user(base, &mut buffer)?;
+            let mut buffer = vec![0u8; io_vec.len()];
+            io_vec.read_exact_from_user(&mut buffer)?;
             buffer
         };
+
         let write_len = file.write(&buffer)?;
```

Comment:
In case that the https://github.com/asterinas/asterinas/pull/702#issuecomment-2100167547 comment is forgotten, add a FIXME comment about atomicity.